_episodes/24.md

```diff
@@ -0,0 +1,412 @@
+    Here we hae two services that each sit behind a load balancer provided and mapped by the Kuberenets cluster.
+   </p>
+
+1. **Kubernetes components and architecture**:
+
+   - Node - The physical machine or VM running a kubelet and container runtime.
+   - Control Plane - The container orchestration layer that exposes the API and 
+   interfaces to define, deploy, and manage the lifecycle of containers.
+   - Cluster - a set of nodes connected to the same control plane.
+   - Pod - single instance of an application, the smallest object in kubernetes.
+
+   <p align="center">
+    <img align="center" width="75%" height="100%" src="/assets/episode/24/components-of-kubernetes.svg"/><br/>
+    Source: https://kubernetes.io/docs/concepts/overview/components/
+   </p>
+
+Kubernetes control plane components:
+
+ - API server that nodes connect with and is the front end for users and 
+ administrators of the cluster.
+ - etcd keystore is a distributed store containing all data used to manage 
+ the cluster
+ - Scheduler that distributes work across nodes and assigns newly created 
+ containers to nodes
+ - Controllers that are the brain behind orchestration and monitors for 
+ nodes going down etc...
+     
+Kubernetes worker nodes components:
+ - container runtime - is the underlying runtime used to manage containers 
+ - kubelet - agent that checks the health and manages the pods running on the node based on the desired state provided in the PodSpec
+ - kube-proxy - network proxy that maintains network rules applied to nodes and allows network access between Pods in a cluster
+
+   You can scale up multiple pods on a single node until the node has no more 
+   resources, at which time a new node needs to be added and pod instances are 
+   distributed between the nodes. 
+
+1. **So how does this relate to Trino?**
+
+ - Out of the box, Kubernetes can do these key things for Trino.
+    - Simple scale up and down (manually tell k8s to startup or kill).
+    - Kubernetes supports failover, meaning that your workers will restart if they die.
+ - Advanced jobs that could exist but not currently in open source.
+    - Auto-scaling via the [Horizontal Pod Autoscaler](https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/) 
+    and custom metrics.
+    - Graceful Shutdowns are hooks that you can add into your cluster that wait 
+    to shut down to avoid a failed call to a node that already shut down.
+     <p align="center">
+        <img align="center" width="75%" height="100%" src="/assets/episode/24/kubernetes-shutdown.svg"/><br/>
+        Source: https://learnk8s.io/graceful-shutdown
+     </p>
+     <p align="center">
+        <img align="center" width="75%" height="100%" src="/assets/episode/24/graceful-shutdown.svg"/><br/>
+        Source: https://learnk8s.io/graceful-shutdown
+     </p>
+     
+1. **What the heck are helm charts then?**
```

Comment:
Maybe a heading?

Reply:
Te be consistent with previous i'll keep this the same